### PR TITLE
fix(java_indexer): assign all diagnostics a corpus

### PR DESCRIPTION
**Pull request by justbuchanan**
when use_compilation_corpus_as_default is enabled

### CODE REVIEW

**Review by jaysachs — APPROVED** [review]

File: kythe/java/com/google/devtools/kythe/analyzers/base/KytheEntrySets.java [hunk 1]
```diff
@@ -202,7 +202,11 @@ public EntrySet newAnchorAndEmit(VName fileVName, Span loc, Span snippet) {
 
   /** Emits and returns a DIAGNOSTIC node attached to no file. */
   public EntrySet emitDiagnostic(Diagnostic d) {
-    return emitDiagnostic(null, d);
+    VName fileVName =
+        useCompilationCorpusAsDefault
+            ? VName.newBuilder().setCorpus(compilationVName.getCorpus()).build()
+            : null;
+    return emitDiagnostic(fileVName, d);
```
Comment (schroederc) [inline kythe/java/com/google/devtools/kythe/analyzers/base/KytheEntrySets.java:209]: This incorrectly attaches the diagnostic to a file VName with an empty path.  The corpus needs to be set in the diagnostics builder

Comment (justbuchanan) [reply kythe/java/com/google/devtools/kythe/analyzers/base/KytheEntrySets.java:213]: Done. PTAL.  The diagnostics that are ending up without a corpus are those where the indexer is unable to find a symbol, so there's no file to attach the diagnostic to.

File: kythe/java/com/google/devtools/kythe/analyzers/java/JavaEntrySets.java [hunk 2]
```diff
@@ -159,8 +159,12 @@ public VName getNode(
...
         logger.atWarning().log("%s", msg);
       }
-      Diagnostic.Builder d = Diagnostic.newBuilder().setMessage(msg);
-      return emitDiagnostic(d.build()).getVName();
+      Diagnostic d = Diagnostic.newBuilder().setMessage(msg).build();
+      VName dVName =
+          getUseCompilationCorpusAsDefault()
+              ? VName.newBuilder().setCorpus(getCompilationVName().getCorpus()).build()
+              : null;
+      return emitDiagnostic(dVName, d).getVName();
```
Comment (schroederc) [inline kythe/java/com/google/devtools/kythe/analyzers/java/JavaEntrySets.java:167]: This is still calling the API that tags a file with a diagnostic.  I think this change needs to be addressed in KytheEntrySets.

**Review by schroederc — CHANGES_REQUESTED** [review]

File: kythe/java/com/google/devtools/kythe/analyzers/base/KytheEntrySets.java (hunk 1 above)
Comment (schroederc) [reply kythe/java/com/google/devtools/kythe/analyzers/base/KytheEntrySets.java:213]: Sorry for any confusion.

`emitDiagnostic(VName, Diagnostic)` will emit a `tagged` edge if the `VName` is non-null.  That's why this change is incorrect.  We will be adding a new `tagged` edge from a `VName` with only a corpus.

I believe this change is meant to only change the corpus on the `Diagnostic` itself which means the change should be located in `emitDiagnostic(VName, Diagnostic)` on both sides of the `fileVName == null` condition.

Comment (justbuchanan) [reply kythe/java/com/google/devtools/kythe/analyzers/base/KytheEntrySets.java:213]: Hmm, when you say to only change the corpus on the `Diagnostic` itself, where is it stored? The `Diagnostic` proto doesn't have any fields to store that AFAIK, only the `ResolvedDiagnostic` does.

Comment (justbuchanan) [reply kythe/java/com/google/devtools/kythe/analyzers/base/KytheEntrySets.java:213]: Can I just set the `CorpusPath` on the `NodeBuilder` in `emitDiagnostic(VName fileVName, Diagnostic d)`?

Comment (schroederc) [reply kythe/java/com/google/devtools/kythe/analyzers/base/KytheEntrySets.java]: Yeah, LGTM.

**Review by schroederc — APPROVED** [review]

**Pull request merged**